### PR TITLE
[FLINK-24768][runtime-web] upgrade Angular and NG-ZORRO versions to 13

### PR DESCRIPTION
**Pull request by yangjunhan**
## What is the purpose of the change

Upgrade Flink Runtime Web's dependencies Angular and NG-ZORRO versions to 13. Close [FLINK-24768](https://issues.apache.org/jira/browse/FLINK-24768).

## Brief change log

Update flink-runtime-web's dependencies and devDependencies.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): yes
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by yangjunhan** [comment]
I have to explicitly add `@ant-design/icons-angular` as a dependency for `flink-runtime-web` even thought it is a dependency for NG-ZORRO.

When I ran `npm install` under node v14.18.X, everything worked fine. But if I ran the former command under node v16.10.X, `@ant-design/icons-angular` did not install as expected. This may be caused by the fact that NG-ZORRO's package.json engines field still uses `^12.20.0 || ^14.0.0`, missing `^16.0.0` (according to [angular version guide](https://gist.github.com/LayZeeDK/c822cc812f75bb07b7c55d07ba2719b3)).

Once NG-ZORRO fixes this issue, I can then safely remove this explicit dependency.

File: flink-runtime-web/README.md [hunk 1]
```diff
@@ -70,6 +70,19 @@ Install *node.js* via:
...
+Once the runtime web's dependencies are updated, please remove the `node_modules` folder manually
+and run the command:
+
+```bash
+cd flink-runtime-web/web-dashboard
+npm install
+```
+
+to install all the up-to-date dependencies **before trying to build the flink-runtime-web through
+Maven**. Otherwise, you may encounter a failure of `npm ci --cache-max=0 --no-save` task while building.
```
Comment (Airblader) [inline flink-runtime-web/README.md:84]: Why does that happen? Does `mvn clean` at least remove `node_modules`? If not we should maybe open a JIRA issue for that. Either way, can you please separate this README change into its own commit since it's unrelated to Angular 13?

File: flink-runtime-web/web-dashboard/src/app/share/customize/job-badge/job-badge.component.spec.ts [hunk 2]
```diff
@@ -22,10 +22,8 @@ import { JobBadgeComponent } from './job-badge.component';
 
 describe('A job badge', () => {
   it('displays the state', async () => {
-    await render(JobBadgeComponent, {
-      componentProperties: {
-        state: 'RUNNING'
-      }
+    await render(`<flink-job-badge [state]="'RUNNING'"></flink-job-badge>`, {
```
Comment (Airblader) [inline flink-runtime-web/web-dashboard/src/app/share/customize/job-badge/job-badge.component.spec.ts:25]: Why is this change necessary? From reading the changelog, it seems that only the `template` key was removed, but we're not using that. The examples on the website still use the same syntax as we did before.

**Review by Airblader — COMMENTED** [review]
Thanks for working on this! Good to see that no real changes are needed. I'll have to run this locally once, but will wait for the CI first.

> Once NG-ZORRO fixes this issue, I can then safely remove this explicit dependency.

Can you please open an issue both against NG-ZORRO and one in the Flink JIRA to track this?

File: flink-runtime-web/web-dashboard/src/app/share/customize/job-badge/job-badge.component.spec.ts (hunk 2 above)
Comment (yangjunhan) [reply flink-runtime-web/web-dashboard/src/app/share/customize/job-badge/job-badge.component.spec.ts:25]: I am referring this [changelog](https://github.com/testing-library/angular-testing-library/releases/tag/v11.0.0). 

image

Comment (Airblader) [reply flink-runtime-web/web-dashboard/src/app/share/customize/job-badge/job-badge.component.spec.ts:25]: But that only says the `template` property has been removed. We didn't use that. I think the way this was written before should still work.

**Comment by yangjunhan** [comment]
> Can you please open an issue both against NG-ZORRO and one in the Flink JIRA to track this?

Thanks for the advice. Here is an issue tracking the problem: [FLINK-25714](https://issues.apache.org/jira/browse/FLINK-25714)

File: flink-runtime-web/README.md (hunk 1 above)
Comment (yangjunhan) [reply flink-runtime-web/README.md]: Yep, I will investigate this issue a bit more before making a statement in README

File: flink-runtime-web/web-dashboard/src/app/share/customize/job-badge/job-badge.component.spec.ts (hunk 2 above)
Comment (yangjunhan) [reply flink-runtime-web/web-dashboard/src/app/share/customize/job-badge/job-badge.component.spec.ts:25]: Yes, I think you are right. I will revert it.

**Comment by yangjunhan** [comment]
@flinkbot run azure

**Review by Airblader — APPROVED** [review]
Thanks! The PR looks good to me; will do some local testing and then proceed with the merge.
